retry/retrier.go

```diff
@@ -0,0 +1,209 @@
+package retry
+
+import (
+	"fmt"
+	"math"
+	"math/rand"
+	"time"
+)
+
+var random = rand.New(rand.NewSource(time.Now().UnixNano()))
```

Comment:
You may want to make it possible to pass in a custom random number generator for testing purposes.

Reply:
good call, will add this!